app/src/main/java/mod/RockAriful/AndroXStudio/GitHubRepoCloner.java

```diff
@@ -0,0 +1,119 @@
+
+package mod.RockAriful.AndroXStudio;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.content.Context;
+import org.eclipse.jgit.api.CloneCommand;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.JGitInternalException;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.eclipse.jgit.lib.ProgressMonitor;
+
+import mod.agus.jcoderz.lib.FileUtil;
+
+public class GitHubRepoCloner {
+
+    private String url;
+    private String name;
+    private String filePath;
+    private String username;
+    private String password;
+
+    public GitHubRepoCloner(String url, String name, String username, String password) {
+        this.url = url;
+        this.name = name;
+        this.filePath = FileUtil.getExternalStorageDir()+"/.sketchware/._temp/";
+        this.username = username;
+        this.password = password;
+    }
+
+    public interface CloneCallback {
+        void onComplete(boolean success, String mesg);
+        void onProgress(int progress);
+    }
+
+    public void cloneRepository(final CloneCallback callback) {
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        final Handler handler = new Handler(Looper.getMainLooper());
+
+        executor.execute(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    CloneCommand clone = Git.cloneRepository();
+                    clone.setURI(url);
+                    clone.setDirectory(new File(filePath, name));
+                    clone.setBare(false);
+                    clone.setCloneAllBranches(true);
+                    clone.setCredentialsProvider(new UsernamePasswordCredentialsProvider(username, password));
+                    clone.call();
+		    _zip(filePath+name+"/DataSource",filePath+name+"/DataSource.swb");
```

Comment:
Shouldn't the SWB file have a more appropriate name than `DataSource.swb`? You can use "SketchwareSource.swb` or something